product_code_unique/models/product.py

```diff
@@ -0,0 +1,14 @@
+# Copyright (C) 2018 - TODAY, Open Source Integrators
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import fields, models
+
+
+class ProductProduct(models.Model):
+    _inherit = 'product.product'
+
+    default_code = fields.Char('Internal Reference', index=True)
+
```

Comment:
Was it needed to write the field definition again ?Cannot we add only the constraint?

Reply:
Thanks for pointing out. Tested and it works the same. This is corresponded.  Unlike with lower odoo version if not redefined changes wont take effect.